### PR TITLE
Add UI for disableAutomaticCheckForUpdates

### PR DESCRIPTION
**Pull request by annakocheshkova**
* [X] Are the files formatted correctly?
* [X] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

image

[AB#75363](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/75363)

### CODE REVIEW

File: Sasquatch/Sasquatch/ViewControllers/MSDistributeViewController.swift [hunk 1]
```diff
@@ -65,14 +66,22 @@ class MSDistributeViewController: UITableViewController, AppCenterProtocol {
     self.edgesForExtendedLayout = []
   }
   
+  @IBAction func checkForUpdateSwitchUpdated(_ sender: UISwitch) {
+      UserDefaults.init().set(sender.isOn ? true : false, forKey: kSASAutomaticCheckForUpdateDisabledKey)
```
Comment (owlstronaut) [inline Sasquatch/Sasquatch/ViewControllers/MSDistributeViewController.swift]: Is `sender.isOn ? true : false` for handling the null case?

File: Sasquatch/Sasquatch.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -1004,7 +1004,7 @@
 		84226C471E88FF4A00798417 /* SasquatchObjC.app */ = {isa = PBXFileReference; explicitFileType = wrapper.application; includeInIndex = 0; path = SasquatchObjC.app; sourceTree = BUILT_PRODUCTS_DIR; };
 		84226C4A1E88FF4A00798417 /* main.m */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.c.objc; path = main.m; sourceTree = "<group>"; };
 		84226C4C1E88FF4A00798417 /* AppDelegate.h */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.c.h; path = AppDelegate.h; sourceTree = "<group>"; };
-		84226C4D1E88FF4A00798417 /* AppDelegate.m */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.c.objc; path = AppDelegate.m; sourceTree = "<group>"; };
+		84226C4D1E88FF4A00798417 /* AppDelegate.m */ = {isa = PBXFileReference; indentWidth = 2; lastKnownFileType = sourcecode.c.objc; path = AppDelegate.m; sourceTree = "<group>"; tabWidth = 2; };
```
Comment (jaeklim) [inline Sasquatch/Sasquatch.xcodeproj/project.pbxproj]: Are you changing something here?

File: Sasquatch/Sasquatch/ViewControllers/MSDistributeViewController.swift [hunk 3]
```diff
@@ -65,14 +67,22 @@ class MSDistributeViewController: UITableViewController, AppCenterProtocol {
...
   }
 
   override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
     switch (indexPath.section) {
-      
+    case 0:
+        switch (indexPath.row) {
+        case 2: appCenter.checkForUpdate()
+        default: ()
+        }
```
Comment (jaeklim) [inline Sasquatch/Sasquatch/ViewControllers/MSDistributeViewController.swift]:
```suggestion
        }

```

File: Sasquatch/SasquatchObjC/AppDelegate.m [hunk 4]
```diff
@@ -119,6 +119,9 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
   if (updateTrack) {
     MSDistribute.updateTrack = updateTrack;
   }
+  if ([[[NSUserDefaults new] objectForKey:kSASAutomaticCheckForUpdateDisabledKey] isEqual:@1]) {
+    //MSDistribute.disableAutomaticCheckForUpdate();
```
Comment (jaeklim) [inline Sasquatch/SasquatchObjC/AppDelegate.m]: This should be ready now.

File: Sasquatch/Sasquatch/ViewControllers/MSDistributeViewController.swift (hunk 1 above)
Comment (annakocheshkova) [reply Sasquatch/Sasquatch/ViewControllers/MSDistributeViewController.swift]: @owlstronaut not sure what null case are you talking about?  `UISwitch` is not an optional.

Comment (owlstronaut) [reply Sasquatch/Sasquatch/ViewControllers/MSDistributeViewController.swift]:
```suggestion
      UserDefaults.init().set(sender.isOn, forKey: kSASAutomaticCheckForUpdateDisabledKey)
```
I'm asking if this is the same thing @annakocheshkova

File: Sasquatch/Sasquatch/ViewControllers/MSDistributeViewController.swift [hunk 5]
```diff
@@ -42,6 +43,7 @@ class MSDistributeViewController: UITableViewController, AppCenterProtocol {
   override func viewDidLoad() {
     super.viewDidLoad()
     self.customized.isOn = UserDefaults.init().bool(forKey: kSASCustomizedUpdateAlertKey)
+    self.autoCheckForUpdate.isOn = UserDefaults.init().bool(forKey: kSASAutomaticCheckForUpdateDisabledKey)
```
Comment (jaeklim) [inline Sasquatch/Sasquatch/ViewControllers/MSDistributeViewController.swift]: Can't you use `standard` here? I don't know why above line uses `init()`.

File: Sasquatch/Sasquatch/ViewControllers/MSDistributeViewController.swift [hunk 6]
```diff
@@ -91,6 +102,6 @@ class MSDistributeViewController: UITableViewController, AppCenterProtocol {
   }
 
   @IBAction func customizedSwitchUpdated(_ sender: UISwitch) {
-    UserDefaults.init().set(sender.isOn ? true : false, forKey: kSASCustomizedUpdateAlertKey)
+    UserDefaults.standard.set(sender.isOn ? true : false, forKey: kSASCustomizedUpdateAlertKey)
```
Comment (owlstronaut) [inline Sasquatch/Sasquatch/ViewControllers/MSDistributeViewController.swift]:
```suggestion
    UserDefaults.standard.set(sender.isOn, forKey: kSASCustomizedUpdateAlertKey)
```

File: Sasquatch/SasquatchSwift/AppDelegate.swift [hunk 7]
```diff
@@ -72,6 +72,9 @@ class AppDelegate: UIResponder, UIApplicationDelegate, MSCrashesDelegate, MSDist
        let updateTrack = MSUpdateTrack(rawValue: updateTrackValue) {
         MSDistribute.updateTrack = updateTrack
     }
+    if UserDefaults.standard.bool(forKey: kSASCustomizedUpdateAlertKey) {
```
Comment (jaeklim) [inline Sasquatch/SasquatchSwift/AppDelegate.swift]: kSASAutomaticCheckForUpdateDisabledKey

**Pull request merged**